### DISCUSSION TITLE
maestral stops running

### DISCUSSION THREAD

**Discussion by gwww** [samschott/maestral#1036] [General]
Similar to a few of the issues that have been reported, maestral just stops running. Two things...

First in the logs I see sequences of messages like the following:

```
2024-04-09 22:58:33 manager INFO: Syncing...
2024-04-09 22:58:36 sync INFO: Syncing ↑ 1/102
2024-04-09 22:58:36 sync INFO: Syncing ↑ 2/102
2024-04-09 22:58:36 sync INFO: Syncing ↑ 3/102
...
2024-04-09 22:58:41 sync INFO: Syncing ↑ 100/102
2024-04-09 22:58:41 sync INFO: Syncing ↑ 101/102
2024-04-09 22:58:41 sync INFO: Syncing ↑ 102/102
2024-04-09 22:58:41 sync INFO: Syncing ↑ 1/1
2024-04-09 22:58:42 sync INFO: Syncing ↑ 1/21
2024-04-09 22:58:42 sync INFO: Syncing ↑ 2/21
...
2024-04-09 22:58:42 sync INFO: Syncing ↑ 20/21
2024-04-09 22:58:43 sync INFO: Syncing ↑ 21/21
2024-04-09 22:58:43 manager INFO: Up to date
2024-04-09 22:58:43 manager INFO: Syncing...
2024-04-09 22:58:45 sync INFO: Syncing ↑ 1/102
2024-04-09 22:58:45 sync INFO: Syncing ↑ 2/102
2024-04-09 22:58:45 sync INFO: Syncing ↑ 3/102
```
The above repeats many times, stops, then seems normal for a day or so, then goes into a pattern like above again. I would describe this as a "sync loop". In the example above, it repeats ~50 times before going back to "normal". 

There is no indication in the log that maestral stops. There are a couple of restarts though. They do not seem to be connected (time wise) to the above logs. Here is one example:
```
2024-04-15 13:50:40 manager INFO: Up to date
2024-04-15 13:53:58 manager INFO: Connecting...
2024-04-15 13:54:44 daemon INFO: Starting daemon
2024-04-15 13:54:44 manager INFO: Connecting...
```
The restart, I assume, is because of what I describe below. Over the period of about 1 week I see three "Starting daemon" messages.

The original problem that got me looking at the logs is described by the title of this discussion. Seemingly randomly, maestral stops running, as in the process is no longer running in the background and the menu bar icon is gone - and clearly syncing has stopped. I usually notice this when someone ask me why I didn't share some file, only to discover maestral has stopped. My question is the start up plist for maestral has this:
```
	<key>RunAtLoad</key>
	<true/>
```
The man page for launchd.plist suggests that this key should be avoided (I did not quite understand the reasoning in the man page). I'm wondering if the following would be more appropriate key to use in the plist?
```
	<key>KeepAlive</key>
        <true/>
```
I have edited my launchd for maestral from `RunAtLoad` to `KeepAlive`. It has been running that way for about 5 days.

I don't know if you want me to create one or more issues for this.

**Comment by gwww** [comment]
I just looked at my log again. In a span of 20 minutes maestral tried to sync 112 times. Isn't Dropbox going to object to that kind of activity?

**Comment by schwerd** [comment]
I continue to have problems with Maestral just stopping. no notice, no warning and no explanation.  have done some testing over the last few days.  it appears to do ok with small files.  but any file over roughly 30 or 40 MB and it chokes and dies.  I restart it. it restarts from ZERO and chokes and dies again. 

why does it not like large files?  this was never an issue before.  for years it's been fine, but now doesn't like big files.  which for me is a big problem. as a photographer I often have large image files.

still hoping we can figure out what's happening and how to adjust

**Comment by schwerd** [comment]
further update as I continue to watch.  Maestral continues to crash repeatedly and often.
many times within 20-30 seconds of restarting the application.
but sometimes it will run for a minute or two.

it appears to be related to the size of files that it's trying to sync.  if there is a file bigger than about 250MB, it crashes reliably.  removing the large file from the syncing folder, and small files sync easily and quickly.

I've tried a number of files testing to see if I can find a filesize, and it appears to be bigger than about 200-250MB.  anything around 1GB or larger (which is many files) it just crashes instantly and all the time.  I've done a number of tests and it's entirely predictable.  big file, it crashes.  remove big file and it syncs flawlessly.

really wish we could get some diagnostic help put in a test version of the app so we could help identify what might be causing this.  for a long time, I thought I was a rare or only outlier, but it's clear there is a larger group of us and the behavior is consistent.

**Comment by schwerd** [comment]
update October 13, 2024

crashing constantly again.  as per before. 

I've done some pretty careful testing.  it seems to crash if there is a file larger than about 50MB. and then won't process any other files in the folder.  20 or 30 files, all small, but if there is one big one, it dies and doesn't do any of them.  definitely chokes on anything bigger than 100MB or even 1GB.  and I have lots of large files. so Maestral has become a huge problem.  I don't know why it crashes now when in the early days, it ran flawlessly regardless of filesize.   I've written an AppleScript that continually makes sure the app is running, and can confirm that if there is a large file, it crashes every single time, but removing the large files and it runs fine on small ones.  I just don't know if it's Dropbox that is causing the problem or Maestral.

I hope there is a way for the developer to figure out why it doesn't like files over 50MB.  

if anyone else can do similar testing to see if they can replicate the results, that would be helpful.

**Reply by thepianoman1** [reply]
> update October 13, 2024

Could you share your  script please? I'm having to deal with conflicted copies all over my workspace because I end up saving while maestral's crashed, one of these days I will accidentally delete the latest version and curse myself...

**Comment by thepianoman1** [comment]
I've had the same issue; Maestral quits/crashes every day, throughout the day. Rarely works for more than an hour. Identical behaviour on both an MBP M1 and MBP M3 running MacOS 15.5 
[maestral.log](https://github.com/user-attachments/files/21034529/maestral.log)

Contrary to the other commenters, the files I edit are generally small,  in the 300 kB range - with a few up to 60 MB.
